Add Vector3 to math utils

ollyau opened the pull request.

ollyau committed:
Add Vector3 to math utils

ollyau committed:
Remove extra include

ollyau committed:
Make empty constructor body more consistent

StevenFrost commented on Source/Utils/Inc/Utils/Math/Vector3.h:
```diff
@@ -0,0 +1,51 @@
+	double Dot( const Vector3& RHS ) const;
+	Vector3 Cross( const Vector3& RHS ) const;
+
+	Vector3 operator +( const Vector3& RHS ) const;
```
Very minor but slight inconsistency with the spacing of operator overloads when compared to those on line 22. I think I've used the no space version elsewhere `Vector3 operator+( const Vector3& RHS ) const;`

StevenFrost commented on Source/Utils/Source/Math/Vector3.cpp:
```diff
@@ -0,0 +1,99 @@
+// -----------------------------------------------------------------------------
+
+Vector3::Vector3()
+	: X( 0 )
```
Can these be 0.0 to match the underlying type?

StevenFrost commented on Source/Utils/Inc/Utils/Math/Vector3.h:
```diff
@@ -0,0 +1,51 @@
+public:
+
+	Vector3();
+	Vector3( double X, double Y, double Z );
```
Doesn't change the behaviour at all but could mark these params as const to indicate that the constructor doesn't change them.

ollyau committed:
Addressing PR #25 comments
- Adjust spacing for operator overloads in header
- Initialize to 0.0 instead of 0 to match type
- Pass parameters as const in constructor for clarity
